File: 0125-valid-palindrome/reverse-comparison/cpp/solution.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        std::string filtered = "";

        for (char character : s) {
            if (std::isalnum(character)) {
                filtered += std::tolower(character);
            }
        }

        return std::equal(filtered.begin(), filtered.end(), filtered.rbegin());
    }
};

File: 0125-valid-palindrome/reverse-comparison/Java/solution.java
class Solution {
    public boolean isPalindrome(String s) {
        StringBuilder filtered = new StringBuilder();

        for (char ele : s.toCharArray()) {
            if (Character.isLetterOrDigit(ele)) {
                filtered.append(Character.toLowerCase(ele));
            }
        }
        
        return filtered.toString().equals(filtered.reverse().toString());
    }
}
